exporter/exporterhelper/internal/consumers.go

```diff
@@ -0,0 +1,49 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal // import "go.opentelemetry.io/collector/exporter/exporterhelper/internal"
+
+import "sync"
+
+type QueueConsumers[T any] struct {
+	queue        Queue[T]
+	numConsumers int
+	callback     func(item QueueRequest[T])
+	stopWG       sync.WaitGroup
+}
+
+func NewQueueConsumers[T any](q Queue[T], numConsumers int, callback func(item QueueRequest[T])) *QueueConsumers[T] {
+	return &QueueConsumers[T]{
+		queue:        q,
+		numConsumers: numConsumers,
+		callback:     callback,
+		stopWG:       sync.WaitGroup{},
+	}
+}
+
+// Start ensures that all consumers are started.
```

Comment:
Also, here is where we still have a bit of "challenging" way to start things. Queue must be started before this, and ended before this. May not be a bad idea for this struct (in a followup PR) to take ownership of starting the queue and ending the queue.

Reply:
Ok, sounds good